systems/subscriber/sim-pool/pkg/server/event.go

```diff
@@ -0,0 +1,48 @@
+package server
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+	epb "github.com/ukama/ukama/systems/common/pb/gen/events"
+	pb "github.com/ukama/ukama/systems/subscriber/sim-pool/pb/gen"
+	"github.com/ukama/ukama/systems/subscriber/sim-pool/pkg/db"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+type SimPoolEventServer struct {
+	simPoolRepo db.SimRepo
+	epb.UnimplementedEventNotificationServiceServer
+}
+
+func NewSimPoolEventServer(simPoolRepo db.SimRepo) *SimPoolEventServer {
+	return &SimPoolEventServer{
+		simPoolRepo: simPoolRepo,
+	}
+}
+func (l *SimPoolEventServer) EventNotification(ctx context.Context, e *epb.Event) (*epb.EventResponse, error) {
+	log.Infof("Received a message with Routing key %s and Message %+v", e.RoutingKey, e.Msg)
+	switch e.RoutingKey {
+	case "event.cloud.simManager.sim.allocation":
```

Comment:
Unused case

Reply:
The SimPool listens to the SimManager when a sim is allocated.That's why we have a listener here

Reply:
yeah  call to function  is missing in case statement